nav2_rviz_plugins/src/selector.cpp

```diff
@@ -148,37 +153,21 @@ void Selector::setProgressChecker()
 }
 
 void
-Selector::timerEvent(QTimerEvent * event)
+Selector::loadPlugins()
 {
-  if (event->timerId() == timer_.timerId()) {
-    if (!plugins_loaded_) {
-      nav2_rviz_plugins::pluginLoader(
-        client_node_, server_failed_, "controller_server", "controller_plugins", controller_);
-      nav2_rviz_plugins::pluginLoader(
-        client_node_, server_failed_, "planner_server", "planner_plugins", planner_);
-      nav2_rviz_plugins::pluginLoader(
-        client_node_, server_failed_, "controller_server", "goal_checker_plugins", goal_checker_);
-      nav2_rviz_plugins::pluginLoader(
-        client_node_, server_failed_, "smoother_server", "smoother_plugins", smoother_);
-      nav2_rviz_plugins::pluginLoader(
-        client_node_, server_failed_, "controller_server", "progress_checker_plugins",
-        progress_checker_);
-
-      plugins_loaded_ = true;
-    }
-
-    // Restart the timer if the one of the server fails
-    if (server_failed_ && !tried_once_) {
-      RCLCPP_INFO(client_node_->get_logger(), "Retrying to connect to the failed server.");
-      server_failed_ = false;
-      plugins_loaded_ = false;
-      tried_once_ = true;
-      timer_.start(200, this);
-      return;
-    }
-
-    timer_.stop();
-  }
+  load_plugins_thread_ = std::thread([this]() {
+    nav2_rviz_plugins::pluginLoader(
+      client_node_, "controller_server", "controller_plugins", controller_);
+    nav2_rviz_plugins::pluginLoader(
+      client_node_, "planner_server", "planner_plugins", planner_);
+    nav2_rviz_plugins::pluginLoader(
+      client_node_, "controller_server", "goal_checker_plugins", goal_checker_);
+    nav2_rviz_plugins::pluginLoader(
+      client_node_, "smoother_server", "smoother_plugins", smoother_);
+    nav2_rviz_plugins::pluginLoader(
+      client_node_, "controller_server", "progress_checker_plugins",
+      progress_checker_);
```

Comment:
It would be nice to track and log the failures here so that we can print something that it failed to connect. I think the `server_failed_` should be maintained